web/client/src/library/pages/ide/IDE.tsx

```diff
@@ -267,13 +274,23 @@ export default function PageIDE(): JSX.Element {
   }, [confirmations])
 
   useEffect(() => {
-    if (models.size > 0) {
+    if (
+      models.size > 0 &&
+      isFalse(planAction.isProcessing) &&
+      isFalse(planAction.isRunningTask)
+    ) {
+      planApply.reset()
+
       void planRun()
     }
   }, [models])
```

Comment:
Is it better to have the dependency be models.size instead of models? Or it doesn't matter?

Reply:
having `models.size` may not catch when some `value` that `models` map holds has changed